### PR TITLE
Fix helm-git cache updates on HA

### PR DESCRIPTION
**Pull request by erikwilson**
Verifies the git catalog cache is at the specified last commit for helm.New.

Makes git catalog detection more robust and adds possibility for explicitly defining the catalog kind (`helm:git` or `helm:http`).

### CODE REVIEW

**Comment by alena1108** [comment]
https://github.com/rancher/rancher/pull/19856

**Comment by dramich** [comment]
lgtm

File: pkg/catalog/helm/setup.go [hunk 1]
```diff
@@ -66,26 +83,27 @@ func newCache(catalog *v3.Catalog) *Helm {
 	}
 }
 
-func (h *Helm) UpdateIfCacheNotExists() error {
-	h.lock()
-	defer h.unlock()
+func GetCatalogKind(catalog *v3.Catalog, localPath string) string {
```
Comment (cjellick) [inline pkg/catalog/helm/setup.go]: this function is only used in this file. should not be exported

**Comment by cjellick** [comment]
lgtm

**Pull request merged**